utils/http.go

```diff
@@ -298,25 +311,36 @@ func (h *HTTP) NewURL(subpath string, values url.Values) *url.URL {
 	return url
 }
 
-// EnsureCert checks cert and key files consistency. It also generates a self-signed cert if it was not specified.
-func (h *HTTP) EnsureCert(defaultPath string) (err error) {
+// EnsureCert checks cert and key files consistency.
+// If the certificate path is not provided in the config,
+// but --insecure-no-tls is not invoked, it'll automatically
+// generaate a self-signed ceritificate and store it in the
+// provided default certificate path.
+func (h *HTTP) EnsureCert(defaultCertPath string) (err error) {
```

Comment:
Why you specified `defaultPath` to `defaultCertPath`? We don't work here with any other types of paths.

Reply:
Just wanted to be extra-explicit about what that path is about. It's also the only argument our function takes, but we're not naming it `a`, because that would be unreadable, right? ;)

Maybe you're right — I don't have a strong preference tbh, I think given that the function is named EnsureCert, it's pretty obvious that an arg named `defaultPath` should contain the cert. 

I'll rename it ;)